C#/C#.sublime-syntax

```diff
@@ -1,278 +1,1634 @@
-      scope: storage.type.source.cs
-  class:
-    - match: '(?=\w?[\w\s]*(?:class|struct|interface|enum)\s+\w+)'
+        - match: '\(|\)'
+          scope: punctuation.definition.expression.cs
+        - match: '&&|\|\||!'
+          scope: keyword.operator.logical.cs
+        - match: '\b(true|false)\b'
+          scope: constant.language.cs
+        - match: '{{name}}'
+          scope: constant.numeric.cs
+        - include: option_done
+    - match: '(else|endif)'
+      scope: keyword.control.preprocessor.cs
+      push: option_done
+    - match: '(error|warning)\s*'
+      scope: keyword.other.preprocessor.cs
+      push: option_message
+    - match: '(region)\s*'
+      scope: storage.type.section.cs
       push:
-        - meta_scope: meta.class.source.cs
-        - match: "}"
-          scope: punctuation.section.class.end.source.cs
-          pop: true
-        - include: storage-modifiers
-        - include: comments
-        - match: (class|struct|interface|enum)\s+(\w+)
-          scope: meta.class.identifier.source.cs
+        - match: ([^/\s][^/]*?)\s*(?=//|/\*|$)
           captures:
-            1: storage.modifier.source.cs
-            2: entity.name.type.class.source.cs
-        - match: ":"
-          push:
-            - match: "(?={)"
-              pop: true
-            - match: '\s*,?([A-Za-z_]\w*)\b'
-              captures:
-                1: storage.type.source.cs
-        - match: "{"
-          scope: punctuation.section.class.begin.source.cs
-          push:
-            - meta_scope: meta.class.body.source.cs
-            - match: "(?=})"
+            1: entity.name.section.cs
+        - include: option_done
+    - match: '(endregion)\s*'
+      scope: storage.type.section.cs
+      push:
+        - match: ([^/\s][^/]*?)\s*(?=//|/\*|$)
+          captures:
+            1: variable.other.section.cs
+        - include: option_done
+
+    - match: '(line)\s+(default|hidden)'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: keyword.other.preprocessor.cs
+      push: option_done
+    - match: '(line)\s*([0-9]*)\s*(".*")?'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: constant.numeric.cs
+        3: string.quoted.double.cs
+      push: option_done
+    - match: '(pragma\s+checksum)\s+(".*")\s+("\{)(.*)(\}")\s+("\{)(.*)(\}")'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: string.quoted.double.cs
+        3: punctuation.definition.hash.begin.cs
+        4: constant.numeric.cs
+        5: punctuation.definition.hash.begin.cs
+        6: punctuation.definition.hash.begin.cs
+        7: constant.numeric.cs
+        8: punctuation.definition.hash.begin.cs
+      push: option_done
+
+    - match: '(pragma)'
+      scope: keyword.other.preprocessor.cs
+      push: option_done
+
+    - match: '(.*)'
+      scope: invalid.illegal.cs
+    - match: $
+      pop: true
+
+  option_message:
+    - meta_content_scope: string.unquoted.cs
+    - include: comments
```

Comment:
It would seem odd that comments should be part of `string.unquoted`. Perhaps instead you should exit if you see `//` or `/*`?